### PR TITLE
kvutils - Treat record_time as a new field  [KVL-1174]

### PR DESCRIPTION
**Pull request by nicu-da**
- backport and integration for https://github.com/digital-asset/daml/pull/11780

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description
- [ ] If you mean to change the status of a component, please make sure you keep [the Component Status page](https://github.com/digital-asset/daml/blob/main/docs/source/support/component-statuses.rst) up to date.

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

**Comment by SamirTalwar-DA** [comment]
I think this is a forward-port.

File: ledger/participant-state/kvutils/src/main/protobuf/com/daml/ledger/participant/state/kvutils/store/state.proto [hunk 1]
```diff
@@ -54,12 +54,13 @@ message DamlCommandDedupKey {
...
 message DamlCommandDedupValue {
   reserved 1; // was record_time
+  reserved 3; // was submission_time
   oneof time {
     // The time until when future commands with the same
     // deduplication key will be rejected due to a duplicate submission.
     // Read-only for backwards compatibility
     google.protobuf.Timestamp deduplicated_until = 2 [deprecated = true];
-    google.protobuf.Timestamp record_time = 3;
+    google.protobuf.Timestamp record_time = 5;
```
Comment (SamirTalwar-DA) [inline ledger/participant-state/kvutils/src/main/protobuf/com/daml/ledger/participant/state/kvutils/store/state.proto:63]: 1 used to be `record_time`. Can we reuse it or is that a terrible idea?

Comment (nicu-da) [reply ledger/participant-state/kvutils/src/main/protobuf/com/daml/ledger/participant/state/kvutils/store/state.proto:63]: I didn't really check what type we had. But I would prefer to continue clean and use a new index just to be on the safe side. What's deleted remains deleted.

Comment (gerolf-da) [reply ledger/participant-state/kvutils/src/main/protobuf/com/daml/ledger/participant/state/kvutils/store/state.proto:63]: It's a terrible idea and reusing it merely gets rid of 1 line of unused protobuf, but brings the potential to cause issues. The clean way out is to never reuse a previously used field.

Comment (SamirTalwar-DA) [reply ledger/participant-state/kvutils/src/main/protobuf/com/daml/ledger/participant/state/kvutils/store/state.proto:63]: I thought you might say that.

Comment (gerolf-da) [reply ledger/participant-state/kvutils/src/main/protobuf/com/daml/ledger/participant/state/kvutils/store/state.proto:63]: Also, see my https://github.com/digital-asset/daml/pull/11780#discussion_r752431994 in the other PR. We probably don't even need this change.

Comment (nicu-da) [reply ledger/participant-state/kvutils/src/main/protobuf/com/daml/ledger/participant/state/kvutils/store/state.proto:63]: Alright, closing this one and purging the field in the other PR.

**Comment by nicu-da** [comment]
Field will be cleaned in the 1.18 release, no need for this PR

**Pull request closed without merging**